### PR TITLE
[NU-1210] Rewritten tests to not use Akka in Component API

### PR DESCRIPTION
**Pull request by ForrestFairy**
## Describe your changes
Replaced NuResourcesTest with NuItTest and NuScenarioConfigurationHelper.

RestAssured library used to test endpoints instead of Akka HTTP.

### Problems
For testing components lists for specific users some components are commented out in file. They appear only if test is run through intellij. 

## Checklist before merge
- [x] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [x] Code is cleaned from temporary changes and commented out lines
- [x] Parts of the code that are not easy to understand are documented in the code
- [x] Changes are covered by automated tests
- [x] Showcase in dev-application.conf added to demonstrate the feature
- [x] Documentation added or updated
- [x] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [x] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+    with NuScenarioConfigurationHelper
+    with NuRestAssureExtensions
+    with NuRestAssureMatchers
+    with RestAssuredVerboseLogging
+    with PatientScalaFutures {
+
+  private val defaultComponentIdProvider: ComponentIdProvider = new DefaultComponentIdProvider(Map.empty)
+
+  "The components endpoint should" - {
+    "return users(test, admin) components list" in {
```
Comment (coutoPL) [inline designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala:42]: I think we should have here at least 2 tests:
1. testing if the service returns components
2. testing if the service returns the same components for `test` and `admin` users

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    with WithMockableDeploymentManager
+    with NuScenarioConfigurationHelper
+    with NuRestAssureExtensions
+    with NuRestAssureMatchers
+    with RestAssuredVerboseLogging
+    with PatientScalaFutures {
+
+  private val defaultComponentIdProvider: ComponentIdProvider = new DefaultComponentIdProvider(Map.empty)
+
+  "The components endpoint should" - {
```
Comment (coutoPL) [inline designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala:41]: "The endpoint for getting components is used"?

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+          .getList("id")
+          .asScala
+          .toList
+
+      responseForAdmin should contain theSameElementsAs adminTrueList
+      responseForTest should contain theSameElementsAs testTrueList
+      responseForAdmin.length > responseForTest.length shouldBe true
+    }
+
+    "return component usages" in {
```
Comment (coutoPL) [inline designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala:180]: "The endpoint for getting component usages is used" (at the root level)?

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+        TestProcessingTypes.Streaming,
+        Some(sourceComponentName),
+        ComponentType.Source
+      )
+
+      given()
+        .auth()
+        .basic("admin", "admin")
+        .and()
+        .pathParam("componentId", componentId.value)
```
Comment (coutoPL) [inline designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala:199]: should we test a case where the `componentId` query param is not passed?

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+        .Then()
+        .statusCode(405)
+        .body(
+          equalTo(
+            s"HTTP method not allowed, supported methods: GET"
+          )
+        )
+    }
+
+    "return 401 when not authenticated" in {
```
Comment (coutoPL) [inline designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala:254]: we should test this case for each endpoint

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+        .auth()
+        .basic("admin", "admin")
+        .when()
+        .get(s"$nuDesignerHttpAddress/api/components/{componentId}/usages")
+        .Then()
+        .statusCode(404)
+        .body(equalTo(s"Component ${badComponent.value} not exist."))
+    }
+
+    "return 405 when not get request" in {
```
Comment (coutoPL) [inline designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala:239]: "when invalid HTTP method is passed"? this is what you wanted to test?

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/api/helpers/NuScenarioConfigurationHelper.scala [hunk 7]
```diff
@@ -39,6 +37,14 @@ trait NuScenarioConfigurationHelper extends ScalaFutures {
   protected implicit val processCategoryService: ProcessCategoryService =
     TestFactory.createCategoryService(ConfigWithScalaVersion.TestsConfig)
 
+  protected def createProcessS(
```
Comment (coutoPL) [inline designer/server/src/test/scala/pl/touk/nussknacker/ui/api/helpers/NuScenarioConfigurationHelper.scala:40]: `createProcessS` - what does the `S` at the end mean?

**Review by coutoPL — CHANGES_REQUESTED** [review]

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+              "The resource requires authentication, which was not supplied with the request"
+            )
+          )
+      }
+    }
+  }
+
+  implicit class ExtractColumn[T <: ValidatableResponse](validatableResponse: T) {
+
+    def extractToList(column: String): List[String] = {
```
Comment (coutoPL) [inline designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala:301]: 1. it's rather `rootKey`, right?
2. and maybe it'd be better if we call it `extractToStringsList` (it's more true ;))

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala (hunk 4 above)
Comment (coutoPL) [reply designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala]: my bad ... didn't notice it's a path param. I thought it's a query param

**Review by coutoPL — CHANGES_REQUESTED** [review]
it's ok in general. Two more, minor things left.

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala (hunk 8 above)
Comment (ForrestFairy) [reply designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala:301]: To be sure, `rootKey` as instead of `column`?

**Review by coutoPL — APPROVED** [review]
Great! LGTM!

**Pull request merged**